### ISSUE TITLE
`vercel dev` fails during preflight

### ISSUE THREAD

**Issue by devatina11yb** [zeit/now#9850]
I get a `TypeError: argument string is required` during preflight while serving locally with `vercel dev`. I believe it is because of the missing `Content-Type` header while parsing the body of the request inside the Serverless Functions helpers.

Oddly enough, it does work without error once deployed in production.

I built a simple project to illustrate the issue.
GitHub repository: https://github.com/devatina11yb/vercel-yoga.

Development endpoint: http://localhost:3000/api.
Production endpoint: https://vercel-yoga.vercel.app/api.

To reproduce the error, you need to run `pnpm run serve:dev`.

Here is the full error message:
```
...\vercel-yoga\node_modules\.pnpm\content-type@1.0.5\node_modules\content-type\index.js:108
    throw new TypeError('argument string is required')
          ^

TypeError: argument string is required
    at parse (...\vercel-yoga\node_modules\.pnpm\content-type@1.0.5\node_modules\content-type\index.js:108:11)
    at parseBody (...\vercel-yoga\node_modules\.pnpm\@vercel+node@2.12.0\node_modules\@vercel\node\dist\serverless-functions\helpers.js:15:26)
    at IncomingMessage.get [as body] (...\vercel-yoga\node_modules\.pnpm\@vercel+node@2.12.0\node_modules\@vercel\node\dist\serverless-functions\helpers.js:80:27)
    at normalizeNodeRequest (file:///.../vercel-yoga/node_modules/.pnpm/@whatwg-node+server@0.7.5/node_modules/@whatwg-node/server/esm/utils.js:76:41)
    at handleNodeRequest (file:///.../vercel-yoga/node_modules/.pnpm/@whatwg-node+server@0.7.5/node_modules/@whatwg-node/server/esm/createServerAdapter.js:82:25)
    at requestListener (file:///.../vercel-yoga/node_modules/.pnpm/@whatwg-node+server@0.7.5/node_modules/@whatwg-node/server/esm/createServerAdapter.js:96:32)
    at genericRequestHandler (file:///.../vercel-yoga/node_modules/.pnpm/@whatwg-node+server@0.7.5/node_modules/@whatwg-node/server/esm/createServerAdapter.js:155:20)
    at Object.apply (file:///.../vercel-yoga/node_modules/.pnpm/@whatwg-node+server@0.7.5/node_modules/@whatwg-node/server/esm/createServerAdapter.js:220:20)
    at Server.<anonymous> (...\vercel-yoga\node_modules\.pnpm\@vercel+node@2.12.0\node_modules\@vercel\node\dist\serverless-functions\serverless-handler.js:20:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

It is a GraphQL API using GraphQL Yoga so there is a possibility that the issue comes from GraphQL Yoga but I still think it is an issue with the Serverless Functions helpers.

**Comment by sundaram2021** [comment]
The error message suggests that there is a problem with the Content-Type header of the incoming request, which is causing a TypeError to be thrown during preflight while serving locally with vercel dev. It seems that the parse function in the content-type module is expecting a string argument, but is not receiving one.

The error is being thrown in the parse function, which is being called by the parseBody function in the @vercel/node module, which is in turn being called by the Serverless Functions helpers.

This issue seems to be related to the use of GraphQL Yoga, as it is a GraphQL API using that framework. However, it could also be an issue with the Serverless Functions helpers themselves.

One possible solution to this issue is to ensure that the incoming requests have a Content-Type header set. You can do this by adding a middleware to your GraphQL Yoga server that sets the Content-Type header to application/json if it is not already set. This can be done using the use method of the GraphQLServer class:

`
const { GraphQLServer } = require("graphql-yoga");

const typeDefs = `
  type Query {
    hello: String!
  }
`;

const resolvers = {
  Query: {
    hello: () => "Hello world!",
  },
};

const server = new GraphQLServer({
  typeDefs,
  resolvers,
});

server.use((req, res, next) => {
  if (!req.headers["content-type"]) {
    req.headers["content-type"] = "application/json";
  }
  next();
});

server.start(() => console.log("Server started"));
`
This middleware will set the Content-Type header to application/json if it is not already set, which should prevent the parse function from throwing a TypeError.

Another possible solution could be to upgrade the version of the content-type module to the latest version, as it could be a bug in an older version. You can try running pnpm update content-type to update the module to the latest version.

**Comment by NekoCharm01** [comment]
Made a discussion about the issue, see https://github.com/orgs/vercel/discussions/2173#discussioncomment-5896787
Unfortunately, GET and HEAD requests cannot have a `Content-Type` header set because these methods do not allow a body in the request, so it is the server that should be responsible for correctly handling the request.

**Comment by U-C-S** [comment]
Can verify if `vercel: "28.17.0"` fixes this issue ??.... Cause I experienced this error upon moving to newer version (like here u using  `28.20.0`)

**Comment by ivandevp** [comment]
FYI, this issue was fixed in https://github.com/vercel/vercel/pull/10077 and it's available now in `vercel@30.2.1`

**Comment by devatina11yb** [comment]
As @ivandevp said, it has now been fixed so I'll close this one.
